enderio-base/src/main/java/com/enderio/base/common/util/AttractionUtil.java

```diff
@@ -0,0 +1,62 @@
+package com.enderio.base.common.util;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.phys.Vec3;
+
+public class AttractionUtil {
+
+    /**
+     * Pulls the entity closer to the given BlockPos. Once it is in inside the collision distance, the function will return true, otherwise it returns false.
+     * @param entity
+     * @param pos
+     * @param speed
+     * @param speed4
+     * @param collisionDistanceSq
+     * @return if the entity is inside the collision distance.
+     */
+    public static boolean hasReachedPos(Entity entity , BlockPos pos, double speed, double speed4, double collisionDistanceSq) {
+       return hasReachedPos(entity, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5, speed, speed4, collisionDistanceSq);
+    }
+    
+    /**
+     * Pulls the entity closer to the given x, y and z coordinates. Once it is in inside the collision distance, the function will return true, otherwise it returns false.
+     * @param entity
+     * @param x
+     * @param y
+     * @param z
+     * @param speed
+     * @param speed4
+     * @param collisionDistanceSq
+     * @return if the entity is inside the collision distance.
+     */
+    public static boolean hasReachedPos(Entity entity , double x, double y, double z, double speed, double speed4, double collisionDistanceSq) {
+        if (entity == null) { //If the entity no longer exists, return false
+            return false;
+        }
```

Comment:
Entities are never null in this context

Reply:
Hmm, true. My intention was to ignore entities that might not existed anymore. Would another check "isremoved" or something like that work?

Reply:
Yes, but maybe cache them with a weak reference and check if it still exists and do an isRemoved check

Reply:
I get why you say that, but I'm not certain it is worth it in this case. The list is made every 5 ticks, and if I want to make a list of weak reference I need to iterate over the already existing list again. I'm not certain if we gain anything by using a weak reference.